### PR TITLE
Disable spock proof generation for migrations

### PR DESCRIPTION
**Pull request by janezpodhostnik**
closes: https://github.com/onflow/flow-go/issues/5919

### CODE REVIEW

**Comment by bluesign** [comment]
Maybe better to encapsulate spock state in an interface, and get as parameter. Here id.bytes() will call fingerprint ( I guess )

File: cmd/util/ledger/migrations/migrator_runtime.go [hunk 1]
```diff
@@ -170,3 +178,13 @@ func NewInterpreterMigrationRuntime(
 		ContractNamesProvider:   env,
 	}, nil
 }
+
+type dummyHasher struct{ size int }
```
Comment (turbolent) [inline cmd/util/ledger/migrations/migrator_runtime.go:182]: Maybe specialize to just the 0 case, ie make this an empty struct

**Comment by janezpodhostnik** [comment]
Since the `id` (of Type RegisterID) only has 2 fields, calling `.Bytes()` probably isn't that problematic.  I will take a look at the profile if that shows up. 

I can make a slightly bigger refactor to get around that part in a separate PR.

File: cmd/util/ledger/migrations/migrator_runtime.go [hunk 2]
```diff
@@ -170,3 +178,13 @@ func NewInterpreterMigrationRuntime(
...
+
+func newDummyHasher(size int) hash.Hasher               { return &dummyHasher{size} }
+func (d *dummyHasher) Algorithm() hash.HashingAlgorithm { return hash.UnknownHashingAlgorithm }
+func (d *dummyHasher) Size() int                        { return d.size }
+func (d *dummyHasher) ComputeHash([]byte) hash.Hash     { return make([]byte, d.size) }
+func (d *dummyHasher) Write([]byte) (int, error)        { return 0, nil }
+func (d *dummyHasher) SumHash() hash.Hash               { return make([]byte, d.size) }
```
Comment (turbolent) [inline cmd/util/ledger/migrations/migrator_runtime.go:189]: Do these have to allocate new slices?

If the size field is kept, would it be possible to allocate once and return the same slice each time?

Given my comment above that the size can maybe be specialized to 0, could these functions just return `nil`?

File: fvm/storage/state/transaction_state.go [hunk 3]
```diff
@@ -187,6 +187,21 @@ func NewTransactionState(
 	}
 }
 
+// NewTransactionStateFromExecutionState constructs a new state transaction directly
+// from an execution state.
+func NewTransactionStateFromExecutionState(
```
Comment (turbolent) [inline fvm/storage/state/transaction_state.go:192]: Maybe call this from `NewExecutionState`, the two functions mostly do the same thing

**Review by turbolent — APPROVED** [review]
Nice!

**Comment by bluesign** [comment]
> Since the `id` (of Type RegisterID) only has 2 fields, calling `.Bytes()` probably isn't that problematic. I will take a look at the profile if that shows up.
> 
> I can make a slightly bigger refactor to get around that part in a separate PR

I think as this is only migration related one time thing, better not to mess with flow-go internals. If this has performance effect, we can just wrap transaction state and shortcut to registers.

https://github.com/onflow/flow-go/compare/master...bluesign:flow-go:deniz/proxy

**Comment by janezpodhostnik** [comment]
That would make me somewhat nervous especially in the migration where we simulate running a transaction, but it could be fine. I will take a closer look 👀.

**Review by fxamacker — APPROVED** [review]

**Comment by janezpodhostnik** [comment]
@bluesign storing register changes separately from transactionState would mean creating child transactionStates and rolling them back would not rollback the changes. This is probably fine since in the migrations we are not partially rolling back (if the migrations are successful). But I would prefer not to do that if it does not show up in the profiles.

I propose we run a test with this PR and see how the profile changes. If `Fingerprint` or something else shows up we can take the approach you outlined, we just have to make sure that we are not committing  any unintentional changes because of  the rollback not properly working.

**Pull request merged**